### ISSUE TITLE
How to disable auto-add to dictionary

### ISSUE THREAD

**Issue by Larx** [AnySoftKeyboard/AnySoftKeyboard#1376]
### Steps to reproduce
1. Type any non-existant word. I have completely disabled the feature that new words should be added to the user dictionary.
2. As soon as I override any proposals given by the app (which more often than not may happen by error) the word is added to the user dictionary.
3. I now can press the remove option in order to remove the word from the dictionary.

### Expected behaviour
If I tell AnySoftKeyboard not to mess automatically with my dictionary it should behave exactly this way.

### Actual behaviour
As soon as I type and confirm a word not in the dictionary, it is added to the dictionary (contrary to the settings). If I don't remove it at once by pressing the "remove" option, it stays in the dictionary.

This happens with all current app versions as provided by the PlayStore when participating in the beta programm. I'm using the German language.

**Comment by terrycloth** [comment]
I just noticed this happening on my phone too.

* AnySoftKeyboard 1.9.1944, version-code 4519 (installed from Google Play Store)
* Android 7.0, with Android security patch level May 1, 2018
* Kernel 3.18.35

**Comment by terrycloth** [comment]
Any thoughts on this issue? I very frequently accidentally hit the B key instead of Space, and so I'm filling up my dictionary with bogus compound words.

**Comment by Larx** [comment]
Yes, it is a completely annoying behaviour. It means that as soon as you mistype something and don't at once hit the remove button, it is in the dictionary. And the wrong words in the dictionary are used for word prediction and that makes matters even worse.

**Comment by terrycloth** [comment]
Also weird is the way it has added things like `https:` to my custom dictionary. I should think the colon would not be considered part of any word.

**Comment by Larx** [comment]
Anybody working on this? This is completely annoying!

**Comment by terrycloth** [comment]
Changing the setting (Language globe icon at the bottom of Keyboard Settings screen > Configure correction behavior > Auto dictionary) to "After some usage" somewhat alleviates the problem. At least incorrect words aren't added to the dictionary unless/until I make the same typo a few times, so it slows down the dictionary clutter. Trying to *disable* Auto dictionary still makes unrecognized words always add to the dictionary though.

**Comment by azzamsa** [comment]
@terrycloth 

> Trying to disable Auto dictionary still makes unrecognized words always add to the dictionary though.

Absolutely, I've disabled everything. But it still add unrecognized words to dictionary.

> to "After some usage" somewhat alleviates the problem. At least incorrect words aren't added to the dictionary unless/until I make the same typo a few times, so it slows down the dictionary clutter. 

I will try this. I think 'do not learn new word' is better, but it doesn't work

The bad news is we can't bulk delete all words in dictionary, we have to do it one by one, or reinstall the app. I have this problem here https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1379

**Comment by azzamsa** [comment]
Watching to [Contributors](https://github.com/AnySoftKeyboard/AnySoftKeyboard/graphs/contributors) section, this project seems one man effort for the most part. 

This is voluntary project, and I believe Menny has other businesses to do. :)
So we have to be patient.

**Comment by Larx** [comment]
I see that, too. However, simply disabling the auto-add function would be way way better than the current situation and seems pretty easy to accomplish.

**Comment by Larx** [comment]
Is there anything going on here? This really sucks! The messed up user dictionary makes ASK more and more unusable, unfortunately.

**Comment by Joppla** [comment]
Any reaction of Menny could be nice.

**Comment by Joppla** [comment]
somebody working on this issue?

**Comment by Larx** [comment]
Would be interesting, yes. But somehow this imho very annoying bug seems to attract few attention :-(.

**Comment by azzamsa** [comment]
Terrycloth's [workaround](https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1376#issuecomment-407740371) help me a bit. It's not so annoying as before.

**Comment by Larx** [comment]
Yes, I know. However I wonder why nothing is being done here. Seems to be a simple problem.

**Comment by azzamsa** [comment]
@menny is actively migrating all language pack to monorepo, he get only little help in term of maintaining but huge userbase. I am sure he will look into the issue when his hand is empty.

**Comment by fang2415** [comment]
I found this thread after getting frustrated by this issue, and found an alternative workaround: Incognito Mode, which is supposed to turn off all learning behavior.  Unfortunately the only documentation I can find for how to turn it on is in the note for commit 253691591ac8cb85564da3ac2358885a3753f5cf:

> Will not learn new words in Incognito mode.
Long-press ENTER to show settings, and pick Incognito Mode

Obviously, the broken option should be disabled, and Incognito Mode should be better documented, but the fact that the desired behavior works somewhere in the app might account for the lack of response on this issue.

**Comment by terrycloth** [comment]
oh, that's a smart workaround.

**Comment by kegelkugel** [comment]
Every time I want to open a bug report for this, the problem disappears. That's because DuckDuckGo put's the keyboard in Incognito Mode. So at least that's a work-around. But then you can't even manually add new words. So please fix this problem!

**Comment by menny** [comment]
@ArenaL5 have fixed this. If I'm not mistaken. Should be fixed for the next release.

**Comment by ArenaL5** [comment]
Yes, commit 8159890377c75a89d841aec62bfd93bc214dffa9 fixes the main issue. You can wait for the next release or try the beta.

I haven't tested the smaller bug in @terrycloth's report, however. I don't know if it's still present:

> Also weird is the way it has added things like `https:` to my custom dictionary. I should think the colon would not be considered part of any word.
